dask/array/core.py

```diff
@@ -869,11 +869,27 @@ def store(sources, targets, lock=True, regions=None, compute=True, **kwargs):
         raise ValueError("Different number of sources [%d] and targets [%d] than regions [%d]"
                          % (len(sources), len(targets), len(regions)))
 
-    updates = [insert_to_ooc(tgt, src, lock=lock, region=reg)
-               for tgt, src, reg in zip(targets, sources, regions)]
-    keys = [key for u in updates for key in u]
+
+    updates = {}
+    keys = []
+    for tgt, src, reg in zip(targets, sources, regions):
+        # if out is a delayed object update dictionary accordingly
+        try:
+            dsk = {}
+            dsk.update(tgt.dask)
+            tgt = tgt.key
+        except AttributeError:
+            dsk = {}
+
+        update = insert_to_ooc(tgt, src, lock=lock, region=reg)
+        keys.extend(update)
+
+        update.update(dsk)
+        updates.update(update)
+
+
```

Comment:
I would expect flake8 to yell at you here for having two empty lines within a function.